modules/dnn/test/test_darknet_importer.cpp

```diff
@@ -140,32 +142,42 @@ class Test_Darknet_nets : public DNNTestLayer
         std::vector<Mat> outs;
         net.forward(outs, outNames);
 
-        std::vector<int> classIds;
-        std::vector<float> confidences;
-        std::vector<Rect2d> boxes;
-        for (int i = 0; i < outs.size(); ++i)
+        for (int b = 0; b < batch_size; ++b)
         {
-            Mat& out = outs[i];
-            for (int j = 0; j < out.rows; ++j)
+            std::vector<int> classIds;
+            std::vector<float> confidences;
+            std::vector<Rect2d> boxes;
+            for (int i = 0; i < outs.size(); ++i)
             {
-                Mat scores = out.row(j).colRange(5, out.cols);
-                double confidence;
-                Point maxLoc;
-                minMaxLoc(scores, 0, &confidence, 0, &maxLoc);
-
-                float* detection = out.ptr<float>(j);
-                double centerX = detection[0];
-                double centerY = detection[1];
-                double width = detection[2];
-                double height = detection[3];
-                boxes.push_back(Rect2d(centerX - 0.5 * width, centerY - 0.5 * height,
-                                       width, height));
-                confidences.push_back(confidence);
-                classIds.push_back(maxLoc.x);
+                Mat out;
+                if (batch_size < 2){
+                    out = outs[i];
+                }else{
+                    // get the sample slice from 3D matrix (batch, box, classes+5)
+                    Range ranges[3] = {Range(b, b+1), Range::all(), Range::all()};
+                    out = outs[i](ranges).reshape(0, {outs[i].size[1], outs[i].size[2]});
```

Comment:
Please replace this c++11 feature to the following equivalent:
```cpp
out = outs[i](ranges).reshape(1, outs[i].size[1]);
```
It makes a `cv::Mat` with number of channels 1 and number of rows `outs[i].size[1]`. Number of columns computes automatically.

source: https://docs.opencv.org/master/d3/d63/classcv_1_1Mat.html#a4eb96e3251417fa88b78e2abd6cfd7d8